### PR TITLE
support package-vc

### PR DESCRIPTION
**Pull request by ksqsf**
package-vc is the new way to install external packages introduced in emacs 29.

this PR consists of three improvements to make blink-search compatible with package-vc:

1. fix illegal version number 2.6-dev, or package-vc refuses to do anything (Could you please check what the expected version is?)
2. make the new load-path visible to the bytecode compiler
3. make interactive commands autoload

### CODE REVIEW

File: blink-search.el [hunk 1]
```diff
@@ -90,7 +90,8 @@
                                                     (if load-file-name
                                                         (file-name-directory load-file-name)
                                                       default-directory)))
-(add-to-list 'load-path blink-search-backend-path t)
+(eval-when-compile
+  (add-to-list 'load-path blink-search-backend-path t))
```
Comment (manateelazycat) [inline blink-search.el]: It's not right, blink-search-backend-path won't add in `load-path` if user not compile blink-search.

**Review by manateelazycat — CHANGES_REQUESTED** [review]

**Comment by manateelazycat** [comment]
It's not necessary to wrap autoload on function, you can use similar technology like lazy-load (https://github.com/manateelazycat/lazy-load) to load function dynamically.

File: blink-search.el (hunk 1 above)
Comment (ksqsf) [reply blink-search.el]: `eval-when-compile` will be evaluated when not being compiled. To quote the doc:

```
In interpreted code, this is entirely equivalent to
progn, except that the value of the expression may be (but is
not necessarily) computed at load time if eager macro expansion
is enabled.
```

**Comment by ksqsf** [comment]
Autoloads are required when the package is used as a package.el-flavor package. Package.el will generate "blink-search-autoloads.el" from autoload annotations, and _only this file_ will be loaded on package initialization, to speed up Emacs startup. (There is a separate feature called `package-quickstart` which also needs correct autoloads.) If there are no autoloads, the user will not be able to call any blink-search command if he uses `package-vc-install`.

I added autoloads blindlessly just now. In retrospect, some are indeed unnecessary. I will update the PR in minutes.

File: blink-search.el (hunk 1 above)
Comment (ksqsf) [reply blink-search.el]: ... I agree this is confusing. Changed to `eval-and-compile`.

**Pull request merged**